### DISCUSSION TITLE
loadbalancer not getting created

### DISCUSSION THREAD

**Discussion by TimHeckel** [mysticaltech/terraform-hcloud-kube-hetzner#91] [Q&A] [answered]
hmmm, when I start from scratch with this config, the loadbalancer is not getting created. Interestingly, I can see some of the hetzner servers not getting powered up as the script is running, so I've been manually turning them on. That could of course be the problem! I will keep tearing down and trying again.

```
location                  = "ash" # change to `ash` for us-east Ashburn, Virginia location
network_region            = "us-east" # change to `us-east` if location is ash
agent_server_type         = "ccx22"
control_plane_server_type = "cpx11"
lb_server_type            = "lb11"

# At least 3 server nodes is recommended for HA, otherwise you need to turn off automatic upgrade (see ReadMe).
servers_num               = 3

# For agent nodes, at least 2 is recommended for HA, but you can keep automatic upgrades.
agents_num                = 3
```

Additionally, I do see this error in the logs - not sure if it's a problem?

```
module.agents[0].hcloud_server.server (remote-exec): Get:13 http://mirror.hetzner.com/debian/security bullseye-security/main amd64 Packages [120 kB]

module.control_planes[0].hcloud_server.server (remote-exec): 02/22 03:20:22 [ERROR] CUID#9 - Download aborted. URI=https://provo-mirror.opensuse.org/tumbleweed/appliances/openSUSE-MicroOS.x86_64-16.0.0-kvm-and-xen-Snapshot20220219.qcow2
module.control_planes[0].hcloud_server.server (remote-exec): Exception: [AbstractCommand.cc:351] errorCode=3 URI=https://provo-mirror.opensuse.org/tumbleweed/appliances/openSUSE-MicroOS.x86_64-16.0.0-kvm-and-xen-Snapshot20220219.qcow2
module.control_planes[0].hcloud_server.server (remote-exec):   -> [HttpSkipResponseCommand.cc:218] errorCode=3 Resource not found
```

**Comment by mysticaltech** [comment]
Check this issue, Hetzner sometimes fails to enter rescue and turns the server off. See https://github.com/kube-hetzner/kube-hetzner/issues/49

**Comment by TimHeckel** [comment]
Thanks for that - yeah I did that and perhaps it helped, but unfortunately the load balancer is not getting created. It's dying with the below failure:

```
null_resource.first_control_plane (remote-exec): Job for k3s.service failed because the control process exited with error code.
null_resource.first_control_plane (remote-exec): See "systemctl status k3s.service" and "journalctl -xeu k3s.service" for details.
null_resource.first_control_plane (remote-exec): Waiting for the k3s server to start...
╷
│ Error: remote-exec provisioner error
│
│   with null_resource.first_control_plane,
│   on master.tf line 64, in resource "null_resource" "first_control_plane":
│   64:   provisioner "remote-exec" {
│
│ error executing "/tmp/terraform_1261623817.sh": Process exited with status 124
```

Seems like the `null` in the `null_resource.first_control_plane` is problematic?

I am going to tear down and try again...

**Reply by mysticaltech** [reply]
No that is expected, we are using a null_ressource on purpose. Just destroy, pull the latest and retry. Don't forget to destroy the lb with hcloud load-balancer delete k3s, and perhaps do the same for the network if needed, see the readme.

**Reply by mysticaltech** [reply]
Now if you do not want to destroy, and still want to recreate a removed LB, just change the traefik_config.yaml manually and reapply.

**Reply by TimHeckel** [reply]
Thanks - I have been doing that...appreciate the tips. It seems like something may be messed up with hetzner's api in us-east now...yeah I will try applying the traefik yaml manually if it still hangs...

actually the process never finishes, so I don't get a new kube-config file, so I cannot use that to apply a traefik.yaml file...

**Reply by mysticaltech** [reply]
Yes, maybe something is up with Hetzner, have noticed a few things too. But what I found useful sometimes is just delete the Hetzner project and start fresh again, grab a new token and retry. It's actually easy, takes 2 minutes to do, and really clears the state deeply.

**Reply by TimHeckel** [reply]
Thanks - just wiped and this time regenned the hetzner token. I'll let you know!

**Reply by mysticaltech** [reply]
Great, if it doesn't work, destroy and retry.

**Reply by TimHeckel** [reply]
Thanks - yeah i will have to keep trying tomorrow eve. I'm done for tonight. It's a little tricky. After sshing in, I can see this error is due to terraform using kubectl to create secrets that already exist...so, i dunno. Must be a timing thing. Thanks again for all your help with this project :)

`error executing "/tmp/terraform_882829115.sh": Process exited with status 124`

**Reply by mysticaltech** [reply]
There are no secrets that are supposed to be involved in the process. It's really weird.

**Reply by mysticaltech** [reply]
Ah I understand.

**Reply by mysticaltech** [reply]
You have to delete the cluster completely Tim!

**Reply by mysticaltech** [reply]
terraform destroy

And in another terminal

hcloud load-balancer delete k3s
hcloud network delete k3s

**Reply by mysticaltech** [reply]
Before starting, make sure that 

hcloud server list 

Returns nothing!

**Comment by TimHeckel** [comment, marked as answer]
Hey - just to close the loop, I pulled the latest from master and retried, and everything worked - I think the issue was a strange combination of things, likely something up with hetzner's api last night. Thanks!

**Reply by mysticaltech** [reply]
Great to hear @TimHeckel! 🚀
